app/src/api/parameters.ts

```diff
@@ -0,0 +1,44 @@
+import { strictParseInt } from '../parse';
+
+import { ApiParamError, ApiParamInvalidFormatError, ApiParamRequiredError } from './errors/api';
+
+
+export function processParam<T>(params: object, paramName: string, processingFn: (x: string) => T, required: boolean = false) {
+    const stringValue = params[paramName];
+
+    if(stringValue == undefined && required) {
+        const err = new ApiParamRequiredError('Parameter required but not supplied');
+        err.paramName = paramName;
+        throw err;
+    }
+
+    try {
+        return processingFn(stringValue);
+    } catch(error) {
+        if(error instanceof ApiParamError) {
+            error.paramName = paramName;
+        }
+        
+        throw error;
+    }
+}
```

Comment:
The `processParam` function handles the parsing of a request parameter. This is in lieu of using an actual library/middleware for parsing parameters.
The function needs to know the name of the parameter, so that it can assemble an appropriate error message if there is any problem with the parsing. Parameters can be specified as required. The actual parsing function can be any function that processes the value to a type different than string, or that validates the format of the string parameter (examples below: parsing an ID, validating against regex). The individual parsing functions should handle undefined parameters gracefully.